.github/workflows/check_issues_state.yaml

```diff
@@ -0,0 +1,44 @@
+name: check issues state (nightly)
+
+on:
+  schedule:
+    - cron: '30 4 * * *'
+  workflow_dispatch: {}
+
+jobs:
+    check_issues_state:
+      runs-on: ubuntu-latest
+      env:
+        GH_TOKEN: ${{ secrets.K8S_TEAM_BOT_GH_PAT }}
+      steps:
+        - name: checkout repository
+          uses: actions/checkout@v4
+          with:
+            fetch-depth: 0
+        - name: check issues
+          run: |
+            HAS_ERROR=false
+            grep -roiIE "//( )*TODO(:){0,1}( )*(https://){0,1}github.com/kong/kubernetes-ingress-controller/issues/[0-9]+" . | while read -r line ; do
+              if [ $(gh issue view --json=state "${line##*/}" | jq '.state' | sed 's/\"//g') = "CLOSED" ]; then
+                echo "closed issue referenced: $line"
+                HAS_ERROR=true
+              fi
+            done
+            if [ $HAS_ERROR = true ]; then
+              exit 1
+            fi
```

Comment:
Maybe we could extract this to `scripts/check-fixed-issues-references.sh`, WDYT?